docs/reference/command_line.rst

```diff
@@ -33,7 +33,7 @@ You could also remove orphan pages with the ``--clean`` option.
 
 .. code-block:: bash
 
-    bin/console sonata:page:create-snapshots --site=all
+    bin/console sonata:page:create-snapshots
```

Comment:
IMO we should keep it. Command without argument(s) should not work.

Reply:
Hi @wbloszyk good point

> 1. Now you can change Page and you will see changes in site when you logged with `SUPER_ADMIN_ROLE` (preview before create new snapshots). This functionality must be keep. You don't change it but I want be sure.

I tested now, change the page and check the page before create a new publication, and no, it does not appear!
I just can see after create a new publication, But I remember this functionality, do you know where is the code that is checking this?

> 2. What with performance and crash possibility? IMHO will be nice to move `BackendInteface` to `SonataPageBundle` with new `symfony/messanger` option only and keep `async`. Restore `async` can be done in `4.x` anyway.

- About `async` functionality, it was requested in the begging of the issue: #1415 for @core23 

> IMO we should keep it. Command without argument(s) should not work.

what do you mean should not work?

- should we not allow run commands without arguments? or this command won't work without this `--site=all`?

Reply:
> I tested now, change the page and check the page before create a new publication, and no, it does not appear!
I just can see after create a new publication, But I remember this functionality, do you know where is the code that is checking this?

I do not know. Probably preview is generated from `Page`. `Snapshot` is  `final cache` version. Can you debug with toolbar and `dump` function  starting from preview page?

> IMO we should keep it. Command without argument(s) should not work.
what do you mean should not work?
should we not allow run commands without arguments? or this command won't work without this --site=all?

`--site=` should be required. Simmilar like `--force` in 'doctrine:schema:update`. 

> About async functionality, it was requested/agreed in the begging of the issue: https://github.com/sonata-project/SonataPageBundle/issues/1415 for @core23

`Async` can be deprecated. We can marge this PR when preview will be fixed.

Reply:
just to update!

I can't reproduce this preview feature even using the 3.x branch! what I saw the different was

- `in release `3.x` using the env=dev`  :  when I update the block generate a new publication
- `in release `3.x` using the env=prod`  :  when I update the block, any publication is generated, But I can't see the changes when I click in the preview

in my branch, the first behaviour doesn't happen, when I update the block content any publication is created, But I looked into the code searching by `dev`, `prod`, listeners and I didn't find anything :(

Reply:
✅ Added the `--site=all` as required
✔️ I tested the preview page, and it has exactly the same behaviour then `3.x` branch, But I can't confirm that you will see the changes before de publication, because I can't reproduce this in any branch, even in other project that I'm using.